### PR TITLE
CRM-18493 Ensure that sort param is valid for use in mysql

### PR DESCRIPTION
**Pull request by seamuslee001**
* [CRM-18493](https://issues.civicrm.org/jira/browse/CRM-18493)

### CODE REVIEW

File: api/v3/utils.php [hunk 1]
```diff
@@ -880,11 +881,31 @@ function _civicrm_api3_get_options_from_params(&$params, $queryObject = FALSE, $
...
+    }
+  }
+  elseif ($sort) {
+    if (CRM_Utils_Rule::mysqlOrderBy($sort)) {
+      $finalSort[] = $sort;
+    }
+    else {
+      throw new API_Exception("Unknown field specified for sort. Cannot order by '$sort'");
+    }
+  }
```
Comment (monishdeb) [inline api/v3/utils.php]: @seamuslee001 you can optimize a bit further by 
```
$options['sort'] = NULL;
if (!empty($sort) {
   foreach ((array) $sort as $s) { 
      if (CRM_Utils_Rule::mysqlOrderBy($s)) {
         $finalSort[] = $s;
       }
       else {
         throw new API_Exception("Unknown field specified for sort. Cannot order by '$s'");
       }
  }
   $options['sort'] = implode(', ', $finalSort);
}

**Comment by colemanw** [comment]
@monishdeb why don't you go ahead and make that change. There's a new feature in GitHub I just noticed: As a project owner you have the ability to push directly into this repo and update the PR even though it's not yours.

**Comment by monishdeb** [comment]
Thanks @colemanw for the update 👍 Updated the PR

**Comment by monishdeb** [comment]
@seamuslee001 did some optimisation, please have a look on the final patch !!

**Pull request merged**